client/lib/post-normalizer/rule-pick-canonical-media.js

```diff
@@ -48,18 +49,6 @@ export default function pickCanonicalMedia( post ) {
 		return post;
 	}
 
-	// jetpack lies about thumbnails/featured_images so we need to make sure its actually an image
```

Comment:
note: I should probably separate out this change.
I don't think we need special handling of `featured_image` since it is always first in `canonical_media` so lodash `find` will pick it first.